### ISSUE TITLE
Other Datasets Problem

### ISSUE THREAD

**Issue by makimon123** [naver-ai/pcmepp#9]
Dear Author, I attempted to apply this method on other datasets; however I have observed that the mu_pdist、sigma_pdist and logits distributions are very concentrated during training , even though the distributions of mean and std themselves seem fine.

**Comment by makimon123** [comment]
and the final training results are relatively poor. I suspect this is because the logits have not been trained well. plz

**Comment by SanghyukChun** [comment]
If you are trying to apply this method for from-scratch training (without any pre-trained weight), it would be difficult to optimize. I recently released a new probabilistic VLM project for from-scratch training:

Probabilistic Language-Image Pre-Training
https://arxiv.org/abs/2410.18857
https://github.com/naver-ai/prolip

There is no full training code yet, but you can easily implement the new loss function:
https://github.com/naver-ai/prolip/blob/89aed36968f055fca897dc51c25156b19412c56c/src/prolip/loss.py#L100-L115

If you need to use PCME++ loss for from-scratch training, you will need additional deterministic loss for a stable convergence. As shown in my new paper
<img width="608" alt="image" src="https://github.com/user-attachments/assets/ecebe6fd-a184-4b0e-99c1-56a390fdd0f9">

**Comment by makimon123** [comment]
Thank you very much for your help! I have tried the new method you suggested, but unfortunately, I am encountering an issue where the values in sigma_pdist remain abnormal and the distribution is very concentrated. This phenomenon has not improved during the training process.

I am wondering if this could be related to the data dimensions. In my dataset, both the mean and log variance are encoded with the shape (Batchsize, Dim), specifically (256, 512). I would appreciate your thoughts on whether the dimensionality could be contributing to this issue.

**Comment by SanghyukChun** [comment]
In my previous experiences, dim 512 is sufficiently large enough in terms of learning uncertainty (less dimension such as 16 may need full covariance, not diagonal covariance). Please note that I have not faced the issue you mentioned in my datasets (including COCO and large-scale web-crawled datasets).

If you still have a problem (even when you simultaneously prob and det losses), it seems that your problem happens because $\mu$ is not trained well. One possible way is first to pre-train your weight only using a deterministic method => You can use anything, such as CLIP or triplet loss. Then fine-tune the pre-trained weight with prob loss.
